pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -716,7 +717,12 @@ public void setInputConsumers(List<Consumer<?>> inputConsumers) {
                         consumer instanceof MultiTopicsConsumerImpl
                                 ? ((MultiTopicsConsumerImpl<?>) consumer).getConsumers().stream()
                                 : Stream.of(consumer))
-                .forEach(consumer -> topicConsumers.putIfAbsent(TopicName.get(consumer.getTopic()), consumer));
+                .forEach(consumer -> {
+                        topicConsumers.putIfAbsent(TopicName.get(consumer.getTopic()), consumer);
+                        if (consumer.getTopic().contains("-partition-")) {
```

Comment:
Please use the `isPartitioned()` in `TopicName` class to determine whether the topic is partitioned.

Reply:
Thanks for review